### DISCUSSION TITLE
Checkmarks

### DISCUSSION THREAD

**Discussion by SterlingStirling** [ultralytics/ultralytics#12781] [Q&A]
Hello! I'm currently working on a project that involves reading documents and trying to detect checkmarks/unfilled checkboxes. They're relatively small, especially compared to the rest of the image. I already tried running a YoloV8 w/ ~50 images of yes's and ~50 images of the empty boxes. It didn't seem to catch any of them, and I was hoping to see if anyone had any advice for technique in terms of how many images of the training data would be necessary/anything like that.

Has anyone done something similar for document scanning? I would've liked to use an OCR, but unfortunately those don't register the checks.

**Comment by pderrenger** [comment]
Hello! Detecting small objects like checkmarks and checkboxes can indeed be challenging. Here are a few tips that might help improve your detection results:

1. **Increase your dataset size**: While 100 images might seem sufficient, small objects often require a larger dataset to capture all the variations. Aim for at least a few hundred images if possible.
2. **Augmentation**: Use image augmentation techniques such as rotation, scaling, and translation to increase the diversity of your training data. This helps the model generalize better.
3. **Modify anchor boxes**: YOLOv8 uses anchor boxes that might be too large for very small objects. Consider customizing the anchor box sizes based on the typical size of checkmarks in your images.
4. **Focus on resolution**: Ensure that the input resolution (`imgsz`) during training and inference is high enough to capture the details of the checkmarks.

Here's a simple example of how you might adjust the training command to use a higher resolution:

```bash
yolo detect train data=your_dataset.yaml model=yolov8n.pt imgsz=640
```

Adjusting these parameters could potentially lead to better performance on small objects like those in your project. Good luck! 🍀
